core-web/libs/utils/src/lib/dot-utils.ts

```diff
@@ -33,3 +35,34 @@ export function generateDotFavoritePageUrl(params: {
         (deviceInode ? `&device_inode=${deviceInode}` : '&device_inode=')
     );
 }
+
+/**
+ * Get the query parameter separator
+ * @param url
+ * @returns
+ */
+export function getQueryParameterSeparator(url: string): string {
+    const regex = /[?&]/;
+
+    return url.match(regex) ? '&' : '?';
+}
+
+/**
+ * This method is used to get the runnable link for the tool
+ * @param url
+ * @returns
+ */
+export function getRunnableLink(url: string, currentPageUrlParams: DotPageToolUrlParams): string {
+    const { currentUrl, requestHostName, siteId, languageId } = currentPageUrlParams;
+
+    url = url
+        .replace('{requestHostName}', requestHostName ?? '')
+        .replace('{currentUrl}', currentUrl ?? '')
+        .replace('{siteId}', siteId ? `${getQueryParameterSeparator(url)}host_id=${siteId}` : '')
+        .replace(
+            '{languageId}',
+            languageId ? `${getQueryParameterSeparator(url)}language_id=${String(languageId)}` : ''
+        );
+
+    return url;
+}
```

Comment:
Did you check the URL api? looks like it could help on this: https://developer.mozilla.org/en-US/docs/Web/API/URL